substrate/executor/src/wasm_executor.rs

```diff
@@ -529,6 +527,21 @@ impl CodeExecutor for WasmExecutor {
 	}
 }
 
+impl CodeExecutor for WasmExecutor {
+	type Error = Error;
+
+	fn call<E: Externalities>(
+		&self,
+		ext: &mut E,
+		code: &[u8],
+		method: &str,
+		data: &[u8],
+	) -> Result<Vec<u8>> {
+		let module = Module::from_buffer(code).expect("all modules compiled with rustc are valid wasm code; qed");
```

Comment:
probably an invalid expectation now, since the code is coming from chain

Reply:
Taken from the previous code base.

So we expect this to sort-of happen? What do we want to do in that case then? Have the result come back with a custom Error? How do expect to recover from invalid-wasm-on-chain (like at all?)?

Can make that a custom error (or error_chain this one), but unsure if this shouldn't be part of a specific PR for that then...

Reply:
we can't really recover in the context of this particular block/state, but we shouldn't panic all the same. should basically just be a "invalid block" that can never be built on and we hope that the consensus algorithm eventually provides us another.

Reply:
doesn't need to be done in this PR though since as you said it was already there. worth making an issue/TODO note in the codebase

Reply:
replace this case with a `?`, however our `WasmExecutor` relies on `rustc`-specific assumptions on quite a few occasions. Not sure we can address those easily without defining/creating some spec about what the features the wasm-code has to offer to be compatible.